Simbad search

ehenneken opened the pull request:
Implementation of object search, like e,g,
```
bibstem:"A&A" object:((LMC OR SMC) AND Andromeda) year:2015
```
(which should return results)

aholachek commented:
hi @ehenneken, so far this pull request translates from object:obj1 --> simbid:123.
But can it translate the reverse way around? e.g. simbid:123 --> object:obj1

Right now, it seems like

ehenneken commented:
Hi Alex

When you submit a query simbid:1234 the actual object search will get skipped because it only looks for "object:" in the query string, and in addition "simbid:1234" is a valid query because "simbid" is a valid Solr field. But maybe I'm misinterpreting your question. The micro service as such can translate in both directions.

I'll be happy to explain in more detail tomorrow.

Edwin

> On May 18, 2016, at 4:56 PM, Alex Holachek <notifications@github.com> wrote:
> 
> hi @ehenneken, so far this pull request translates from object:obj1 --> simbid:123.
> But can it translate the reverse way around? e.g. simbid:123 --> object:obj1
> 
> Right now, it seems like
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

aholachek commented:
Ah ok, I just checked out your branch and if you make a search, then reload the page, you no longer see object:foo, you see simbid:123. Is that ok, or was the intention to hide this from the user?

adsabs-beer-user commented:
Strange! The original query should remain in the query box. I'll look at it tomorrow!

> On May 18, 2016, at 5:17 PM, Alex Holachek <notifications@github.com> wrote:
> 
> Ah ok, I just checked out your branch and if you make a search, then reload the page, you no longer see object:foo, you see simbid:123. Is that ok, or was the intention to hide this from the user?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>

ehenneken commented:
Hi @aholachek 

I thought if I do the following to store the original query (e.g. "object:Andromeda") like
```
https://github.com/ehenneken/bumblebee/blob/f5a9cfda326d5271d484230a53fb2b4acffa07b8/src/js/widgets/search_bar/search_bar_widget.js#L489```
and then use it like
```
https://github.com/ehenneken/bumblebee/blob/f5a9cfda326d5271d484230a53fb2b4acffa07b8/src/js/widgets/search_bar/search_bar_widget.js#L322
```
the query box would display that original query "object:Andromeda" (rather than what it was replaced by, i.e. "simbid:1575544")

ehenneken commented:
The SIMBAD search machinery has been moved to the `query_mediator` and the URL keeps the original `object:....` search in it. But with a hard reload the search box still gets `simbid:...` in it. Working on that...

aholachek commented:
Hi @ehenneken, there is some unexpected behavior from the object api parser. When I search "simbid0(LMC OR SMC) planet" the api endpoint returns the string "simbid0(LMC OR SMC) planet" which bumblebee cannot handle. I'm not sure why the : gets replaced with a 0.

Would you mind squashing all object search commits into 1 and rebasing on top of the current master?

ehenneken commented:
the query should be "object:'LMC OR SMC'" rather than "simbid:'LMC OR SMC'"

I'll have a look, probably not before Thursday, and squash all the commits.

ehenneken commented:
Hi @aholachek a query `object:"SMC OR LMC" bibstem:"ApJ"` works as expected, and a query `object:"SMC OR LMC" planet` also in the sense that it returns 0 results (unlikely you'll find something about a planet in papers about the Small and Large Magellanic Clouds (galaxies). I'll let you know when I've squashed the commits and rebased with master.

aholachek commented:
Oops I copy pasted the wrong thing, the query object:(LMC OR SMC) planet is throwing an error of "INVALID_SYNTAX: Syntax Error: Unknown function "simbid0" please use this tool to fix it." in your current branch

aholachek commented on src/js/apps/discovery/navigator.js:
```diff
@@ -325,7 +325,12 @@ define([
           var q = app.getObject('AppStorage').getCurrentQuery();
-          var route = '#search/' + queryUpdater.clean(q).url();
+          if (q && q.get("__original_url")) {
+            var route = '#search/' + q.get("__original_url");
```
Would there be a way to call queryUpdater.clean() on the __original_url before you store it? Otherwise you could potentially put a very long url in the address bar, full of parameters that are supposed to stay hidden. (E.g. if you search for object:LMC then apply a filter using the author network, the url will be full of __bigquery parameters)

aholachek commented on src/js/widgets/search_bar/search_bar_widget.js:
```diff
@@ -316,7 +320,11 @@ define([
         },
 
         setFormVal: function(v) {
```
Since several other functions depend on using setFormVal, I think this change to the function will prevent other things in the search widget (quick add buttons, query mediator) from working correctly (but I could be wrong)

aholachek commented on test/mocha/js/widgets/search_bar_widget.spec.js:
```diff
@@ -398,6 +402,21 @@ define([
     });
 
+    it("check if the SIMBAD 'object:' search is properly translated into 'simbid:' search", function() {
+      var widget = _widget();
```
since most of the translation logic was moved to the query_mediator, could you add tests in query_mediator.spec.js?
I'm possibly (probably) confused but it seems like this particular test isn't testing anything object-query specific, just showing that a start_search event is emitted when the submit button is clicked
